diffkemp/compare.py

```diff
@@ -81,163 +166,152 @@ def compare(args):
             fun_result = functions_diff(
                 mod_first=old_fun_desc.mod, mod_second=new_fun_desc.mod,
                 fun_first=fun, fun_second=fun,
-                glob_var=glob_var, config=config,
-                prev_result_graph=result_graph, function_cache=cache,
-                module_cache=module_cache, modules_to_cache=modules_to_cache)
-            result_graph = fun_result.graph
-
-            if fun_result is not None:
-                if args.regex_filter is not None:
-                    # Filter results by regex
-                    _filter_result_by_regex(args.regex_filter, fun_result)
-
-                result.add_inner(fun_result)
-
-                # Printing information about failures and non-equal functions.
-                if fun_result.kind in [Result.Kind.NOT_EQUAL,
-                                       Result.Kind.UNKNOWN,
-                                       Result.Kind.ERROR] or config.full_diff:
-                    group_printed = _print_fun_result(fun_result, config,
-                                                      output_dir, fun,
-                                                      group_dir, group_name,
-                                                      args, old_fun_desc,
-                                                      group_printed)
+                glob_var=glob_var, config=self.config,
+                prev_result_graph=self.result_graph, function_cache=self.cache,
+                module_cache=self.module_cache,
+                modules_to_cache=self.modules_to_cache)
+            self.result_graph = fun_result.graph
+
+            self._handle_fun_result(fun_result, fun, old_fun_desc)
+            self._cleanup_modules(old_fun_desc, new_fun_desc)
+            return
+
+        @staticmethod
+        def _modules_exist(old_fun_desc, new_fun_desc):
+            return old_fun_desc.mod is not None and \
+                   new_fun_desc.mod is not None
+
+        def _handle_missing_module(self, fun, old_fun_desc):
+            fun_result = Result(Result.Kind.UNKNOWN, fun, fun)
+            self.result.add_inner(fun_result)
+            self._print_fun_result(fun_result, fun, old_fun_desc)
+            return
+
+        def _handle_fun_result(self, fun_result, fun, old_fun_desc):
+
+            if self.args.regex_filter is not None:
+                # Filter results by regex
+                self._filter_result_by_regex(fun_result)
+
+            self.result.add_inner(fun_result)
+
+            # Printing information about failures and
+            # non-equal functions.
```

Comment:
I think it would be better if the wrapping of the code/comments
- was removed
- or wrapped later

if possible (check other locations).
 


```suggestion
            # Printing information about failures and non-equal functions.
```